lib/jnpr/junos/device.py

```diff
@@ -412,6 +422,99 @@ def open(self, *vargs, **kvargs):
             originating ``Exception`` is assigned as ``err._orig``
             and re-raised to the caller.
         """
+        #
+        #  PRIVATE Functions
+        #
+
+        def _get_connected_slot():
+            """return connected slot. In case of error return None"""
+
+            rsp = self.rpc.get_chassis_inventory()
+            if rsp.tag == 'error':
+                raise RuntimeError('Could not determine RE slot id')
+            if rsp.tag == 'output':
+                # this means that there was an error; due to the
+                # fact that this connection is not on the master
+                # @@@ need to validate on VC-member
+                self._reRole = "backup"
+            else:
+                self._reRole = "master"
+
+            rsp = self.rpc.get_route_engine_information()
+            if rsp.tag == 'error':
+                raise RuntimeError('Could not determine RE slot id')
+            re_list = rsp.xpath('.//route-engine')
+            if len(re_list) == 1:
+                return "RE0"
```

Comment:
This is an incorrect assumption. While uncommon, a chassis can have a single RE installed in slot 1. If there's only one RE, you should still look at the <slot> tag to determine the correct slot.